# Size fractionation informs microbial community composition and interactions in the eastern tropical North Pacific ocean

## Madeleine A. Thompson, David L. Valentine, Xuefeng Peng

Raw reads generated in this study are available at the National Center for Biotechnology Information (NCBI) under BioProject PRJNA911303. 

### Summary 
Marine microorganisms are key drivers of biogeochemical cycles in the world’s oceans, including oxygen minimum zones (OMZs). Using a metabarcoding survey of the 16S rRNA gene, we investigated bacterial and archaeal communities, as well as their interactions with fungi, at the coastal, offshore, and peripheral OMZ of the eastern tropical North Pacific. Water samples were collected along a sharp vertical oxygen gradient, and large volumes (23 - 69 L) were filtered through three size fractions, 0.22 μm, 2 μm, and 22 μm. Differential abundance analysis revealed that the changes in community composition along the oxygen gradient were primarily driven by Planctomycetota, Bacteroidota, Verrucomicrobiota, and Gammaproteobacteria, most of which were known for their high hydrolytic potentials and usually associated with the large particle-associated size fraction (> 22 μm). Network analyses using a pipeline that removes indirect interactions resulted in fewer nodes and edges than in previous studies. By including fungal community data collected from the same expedition, network analyses identified interactions between fungi (Rhodotorula and Cladosporium) and prokaryotes in the particle-associated fractions. Putative keystone species identified by network analyses included SAR11, Prochlorococcus, and Nitrospina in the free-living fraction and Vibrionaceae, Phycisphaeraceae, and Flavobacteriaceae in large particle-associated fractions. A Cladosporium ASV was identified as a putative keystone species in the small-particle-associated communities, highlighting previously overlooked interactions between fungi and prokaryotes in the ocean. Our findings demonstrate the utility of large-volume size-fractioned filtration in providing novel insights into marine microbial ecology.

### Figures 
Figure 1 was made using Ocean Data View. The World Ocean Atlas 2018 data was used to image ocean oxygen data at 150 m at the study site in March 2018. 
Figure 2 was made using Graph Prism from the CDT and nutrient data collected during sampling. 
Figure 3 was made using R from the ASV (ZOTU) data and metadata. 
Figure 4 was made using R from the ASV (ZOTU) data and metadata. 
Figure 5 was made using R from the ASV (ZOTU) data and metadata. 
Figure 6 was made using MENAP from the ASV (ZOTU) data (http://ieg4.rccc.ou.edu/MENA/main.cgi) and was refined in Cytoscape. 
Figure 7 was made using MENAP from the ASV (ZOTU) data (http://ieg4.rccc.ou.edu/MENA/main.cgi) and was refined in Cytoscape. 
Figure 8 was made using MENAP from the ASV (ZOTU) data (http://ieg4.rccc.ou.edu/MENA/main.cgi) and was refined in Cytoscape. 


